pkg/postage/listener/listener_test.go

```diff
@@ -0,0 +1,446 @@
+package listener_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/postage/listener"
+)
+
+var hash common.Hash = common.HexToHash("ff6ec1ed9250a6952fabac07c6eb103550dc65175373eea432fd115ce8bb2246")
+var addr common.Address = common.HexToAddress("abcdef")
+
+var postageStampAddress common.Address = common.HexToAddress("eeee")
+var priceOracleAddress common.Address = common.HexToAddress("eeef")
+
+var (
+	postageStampABI    = parseABI(listener.PostageStampABI)
+	priceOracleABI     = parseABI(listener.PriceOracleABI)
+	createdTopic       = postageStampABI.Events["BatchCreated"].ID
+	topupTopic         = postageStampABI.Events["BatchTopUp"].ID
+	depthIncreaseTopic = postageStampABI.Events["BatchDepthIncrease"].ID
+	priceUpdateTopic   = priceOracleABI.Events["PriceUpdate"].ID
+)
+
+func TestListener(t *testing.T) {
+	// test that when the listener gets a certain event
+	// then we would like to assert the appropriate EventUpdater method was called
+	t.Run("create event", func(t *testing.T) {
+		c := createArgs{
+			id:               hash[:],
+			owner:            addr[:],
+			amount:           big.NewInt(42),
+			normalisedAmount: big.NewInt(43),
+			depth:            100,
+		}
+
+		ev, evC := newEventUpdaterMock()
+		mf := newMockFilterer(
+			WithFilterLogEvents(
+				c.toLog(),
+			),
+		)
+		listener := listener.New(logging.New(ioutil.Discard, 0), mf, postageStampAddress, priceOracleAddress)
```

Comment:
maybe pull the logger into a test level variable and reuse across tests

Reply:
fixed